README.md

```diff
@@ -1,16 +1,1605 @@
+          }
+        }
+      },
+      "subject": "Coffee",
+      "body": {
+        "await/ok": {
+          "ucan/task": {
+            "/": "bafyreif7wiz4mz3ojmmvrz2p5m6ay5nqbi7ghwmheccv5rqgjtkrk5wyi4"
+          }
+        }
+      }
+    },
+    "prf": [
+      { "/": "bafyreihvee5irbkfxspsim5s2zk2onb7hictmpbf5lne2nvq6xanmbm6e4" }
+    ]
+  }
+}
+```
+
+By analogy, above examples can be interpreted roughly as follows:
+
+```js
+const createDraft = crud.create("https://example.com/blog/posts", {
+  payload: {
+    title: "How UCAN Tasks Changed My Life",
+    body: "This is the story of how one spec changed everything...",
+  },
+})
+
+const getEditors = crud.read("https://example.com/users/editors")
+
+const notify = msg.send("mailto:akiko@example.com", {
+  to: (await createDraft).ok,
+  subject: "Coffee",
+  body: (await getEditors).ok,
+})
+```
+
+While a [Await] MAY be substituted for any field in a [Task], substituting the `do` field is NOT RECOMMENDED. The `do` field is critical in understanding what kind of action will be performed, and schedulers SHOULD use this fields to grant atomicity, parallelize tasks, and so on.
+
+After resolution, the [Invocation] MUST be validated against the [Authorization] and linked UCAN proofs by the [Executor]. A Promise resolved to an [Invocation] that is not backed by a valid UCAN MUST NOT be executed, and SHOULD return an unauthorized error to the user. A Promise resolved to an [Invocation] with the [Authorization] that does not include invoked [Task] MUST NOT be executed, and SHOULD return an unauthorized error to the user.
+
+Promises MAY be used across [Invocation]s with a same [Authorization], or across [Invocation]s with different [Authorization] and MAY even be across multiple Invokers and Executors. As long as the invocation can be resolved, it MAY be promised. This is sometimes referred to as ["promise pipelining"](http://erights.org/elib/distrib/pipeline.html).
+
+
+## 8.1 Promise
+
+A `Promise` describes eventual output `Result` of the [Invocation].
+
+### 8.1.1 Schema
+
+```ipldsch
+type Promise<In> union {
+  | &Invocation<In>    ("ucan/invocation")
+  | &Task<In>          ("ucan/task")
+} representation keyed
+```
+
+#### 8.1.2 Variants
+
+##### 8.1.2.1 Invocation Promise
+
+A `Promise` of the [Invocation] MUST be a link to that invocation wrapped in the `"ucan/invocation"` tag.
+
+```ts
+{
+  "ucan/invocation": {
+    "/": "bafyreihdw5dpnggixaee3rwn5kr3vhvfvokj3hi24uk2swp4nizyjdhdeq"
+  }
+}
+```
+
+##### 8.1.2.2 Task Promise
+
+An `Promise` of the [Invocation] that shares [Authorization] with a [Task] referecing it MUST be a link to the invocation [Task] wrapped in the `"ucan/task"` tag.
+
+Note that [Task]s with the same [Authorization] will be unable to referece [Invocation] since [Authorization] CAN be created after all the [Task]s being authorized. The `"ucan/task"` tagged Promise MAY be used in such cases.
```

Comment:
```suggestion
Note that [Task]s with the same [Authorization] will be unable to reference [Invocation] since [Authorization] CAN be created after all the [Task]s being authorized. The `"ucan/task"` tagged Promise MAY be used in such cases.
```